packages/shared/src/components/squads/SquadHeaderMenu.tsx

```diff
@@ -76,11 +82,29 @@ export default function SquadHeaderMenu({
         },
         label: 'Feedback',
       },
-      canDeleteSquad
-        ? { Icon: TrashIcon, onClick: onDeleteSquad, label: 'Delete Squad' }
-        : { Icon: ExitIcon, onClick: onLeaveSquad, label: 'Leave Squad' },
     ];
 
+    if (canDeleteSquad) {
+      list.push({
+        Icon: TrashIcon,
+        onClick: onDeleteSquad,
+        label: 'Delete Squad',
+      });
+    }
+
+    if (
+      squad.currentMember &&
+      squad.currentMember.role !== SourceMemberRole.Admin
+    ) {
+      list.push({
+        Icon: ExitIcon,
+        onClick: () => {
+          onLeaveSquad();
+        },
```

Comment:
```suggestion
        onClick: onLeaveSquad,
```

This should do right

Reply:
Typescript complaining.